src/libraries/Microsoft.PowerFx.Connectors/Execution/HttpFunctionInvoker.cs

```diff
@@ -160,33 +163,56 @@ public async Task<FormulaValue> DecodeResponseAsync(HttpResponseMessage response
         {
             // $$$ Do we need to check response media type to confirm that the content is indeed json?
             var json = await response.Content.ReadAsStringAsync();
+            var statusCode = (int)response.StatusCode;
+            var message = statusCode == 200 ? string.Empty : $"Connector call failed ({response.StatusCode}), request {response.RequestMessage.RequestUri}, request headers {FormatHeaders(response.RequestMessage.Headers)}";
 
-            if (!response.IsSuccessStatusCode)
+            return statusCode switch
             {
-                var msg = $"Connector call failed {response.StatusCode}): " + json;
+                int i when i >= 300 && i < 400 => FormulaValue.NewError(new ExpressionError() { Kind = ErrorKind.ConnectorWarning, Severity = ErrorSeverity.Warning, Message = message }),
```

Comment:
>FormulaValue.NewError [](http://example.com/codeflow?start=50&length=21)

I think we need a new overload of NewError that accepts the error type. 
FormulaValue.NewError() currently just uses blank() type.